common/persistence/sql/queue_v2.go

```diff
@@ -315,13 +332,22 @@ func (q *queueV2) getQueueMetadata(
 	tc sqlplugin.TableCRUD,
 	queueType persistence.QueueV2Type,
 	queueName string,
-) (*QueueV2Metadata, error) {
+) (*persistencespb.Queue, error) {
 
 	filter := sqlplugin.QueueV2MetadataFilter{
 		QueueType: queueType,
 		QueueName: queueName,
 	}
-	metadata, err := tc.SelectFromQueueV2Metadata(ctx, filter)
+	var (
+		metadata *sqlplugin.QueueV2MetadataRow
+		err      error
+	)
+	switch tc.(type) {
+	case sqlplugin.Tx:
+		metadata, err = tc.SelectFromQueueV2MetadataForUpdate(ctx, filter)
+	default:
+		metadata, err = tc.SelectFromQueueV2Metadata(ctx, filter)
+	}
```

Comment:
This looks a little bit brittle to me because, everywhere we use `SelectFromQueueV2Metadata`, we'll need to check the type of the tx to see if it's a transaction or not. How about just exposing one `SelectFromQueueV2Metadata` method, and the implementation of that itself checks if we're in a transaction or not? That way, it's impossible for clients to misuse. I'll just make a follow-up though because this isn't strictly necessary, and I'm not 100% sure if it'll even be better. Also, we need this fix merged ASAP.